src/extractors/jsxtag-gettext.js

```diff
@@ -0,0 +1,57 @@
+import * as t from 'babel-types';
+import gettext from './tag-gettext';
+import { PO_PRIMITIVES } from '../defaults';
+import { template2Msgid, msgid2Orig, hasExpressions,
+    ast2Str, getQuasiStr, strToQuasi, dedentStr } from '../utils';
+import { NoTranslationError } from '../errors';
+
+const { MSGSTR } = PO_PRIMITIVES;
+const NAME = 'jsxtag-gettext';
+
+function match(node, context) {
+    return t.isTaggedTemplateExpression(node) && node.tag.name === context.getAliasFor(NAME);
+}
+
+function resolveDefault(nodePath, poData, context) {
```

Comment:
Sorry for this part, because it's not documented yet, and recently I have changed resolve and extract functions signatures.

in general, you should implement `match`, `validate`, `getMsgid`, `extract` and `resolve`, `resolveDefault` functions. Those all can be covered with simple unit tests( checkout example here - https://github.com/c-3po-org/babel-plugin-c-3po/blob/master/tests/unit/test_tag_gettext_extractor.js).
 
Also, you can check an example of how `tag-gettext` is implemented (in a new way) here https://github.com/c-3po-org/babel-plugin-c-3po/blob/master/src/extractors/tag-gettext.js. 

You can start with simple functions as `match` `getMsgid` and `validate`. If I will have a chance I will also contribute to your PR today or tomorrow.

Or you can just leave this task for me because this part is not documented yet, I will try to implement and document it as soon as I will have a chance)